aptos-move/block-executor/src/scheduler.rs

```diff
@@ -330,6 +337,50 @@ impl Scheduler {
 
         SchedulerTask::NoTask
     }
+
+    /// Set the done marker. Should only be called when there is a module r/w intersection.
+    pub fn set_done(&self) {
+        self.done_marker.store(true, Ordering::SeqCst);
+    }
+
+    /// Checks whether the done marker is set. The marker can only be set by 'check_done' and 'set_done'.
+    pub fn done(&self) -> bool {
+        self.done_marker.load(Ordering::Acquire)
+    }
+
+    /// Whichever thread that first sets this atomic variable to false (and thus returns true)
+    /// will be in charge of waking up the pending threads.
+    pub fn is_first(&self) -> bool {
+        self.is_first.fetch_and(false, Ordering::SeqCst)
+    }
+
+    /// When the parallel execution encountered a module r/w intersection and can abort earlier, some of the threads
+    /// may still be working on execution, and waiting for dependency (indicated by the condition variable `condvar`).
+    /// Therefore the commit thread needs to wake up all such pending threads, by sending notification to the condition
+    /// variable and setting the lock variables properly.
+    pub fn resolve_condvar(&self, txn_idx: TxnIndex) {
+        let status = self.txn_status[txn_idx].lock();
+        {
+            // Only transactions with status Suspended or ReadyToExecute may have the condition variable of pending threads.
+            if let TransactionStatus::Suspended(_, condvar) = &*status {
```

Comment:
do a match, extract condvar in both cases, e.g.
match &*status {
TransactionStatus::Suspended(_, condvar) | TransactionStatus::ReadyToExecute(_, Some(condvar)) =>
{
// inside here handle it, e.g. set the status to Terminated.
}
}